src/manager/manager6.c

```diff
@@ -15,7 +12,7 @@ void sub_08057CB4(Manager6* this) {
         this->manager.action = 1;
         this->warpList = GetCurrentRoomProperty(this->manager.unk_0a);
         if (!this->warpList) {
-            DeleteManager((Entity*)this);
+            DeleteManager((Manager*)this);
```

Comment:
Should this be a cast or
```suggestion
            DeleteManager(&this->manager);
```
?

Reply:
this is part of the polymorphism problem, so I'd move this question to another day in another PR once we know how we wanna handle polymorphism

Reply:
I would go with the suggestion, since this is what OOT does for actors.

Reply:
I'll change it
I'll wait with the commit to see if there are other changes to keep the number of commits low